lib/postgrex/protocol.ex

```diff
@@ -201,10 +209,10 @@ defmodule Postgrex.Protocol do
     case :gen_tcp.recv(sock, 1, timeout) do
       {:ok, <<?S>>} ->
         ssl_connect(s, opts)
-      {:ok, <<?N>>} ->
-        error(%Postgrex.Error{message: "ssl not available"}, s)
+      {:ok, reason} ->
+        error(Postgrex.Error.exception(:ssl, "not available", reason), s)
```

Comment:
I think this change is invalid because `<<?N>>` is not a reason, the previous "ssl not available" is a correct message. If postgres sends anything other than `<<?N>>` we are fine to crash as it shouldnt happen.